cameraserver/src/main/java/edu/wpi/first/vision/VisionRunner.java

```diff
@@ -77,7 +77,7 @@ public VisionRunner(VideoSource videoSource, P pipeline, Listener<? super P> lis
   public void runOnce() {
     Long id = CameraServerSharedStore.getCameraServerShared().getRobotMainThreadId();
 
-    if (id != null && Thread.currentThread().getId() == id.longValue()) {
+    if (id != null && Thread.currentThread().getId() == id) {
```

Comment:
Does this change the boxing behaviour here? I would've thought this would unbox for the comparison.

Reply:
This does unbox `id` for the comparison because `Thread.currentThread().getId()` returns a primitive.

I am not sure I answered your question though.

Reply:
Ah, I somehow missed that `Thread#getId` returns a primitive. Never mind me.